### ISSUE TITLE
azmq release

### ISSUE THREAD

**Issue by ahundt** [zeromq/azmq#101]
It may be wise to select outstanding issues that are important, resolve them, then create a release of azmq. This makes it easier to incorporate azmq into package management systems.

For example, I've created a [small Homebrew formula](https://github.com/ahundt/homebrew-robotics/blob/master/azmq.rb), but I believe one of the preferred criteria is to have a release. I've also created an [issue in homebrew for incorporating the formula](https://github.com/Homebrew/homebrew/issues/42836).

**Comment by oliora** [comment]
It is nice idea. Later we can create deb and rpm packages as well as homebrew and biicode.

**Comment by rodgert** [comment]
What would you consider as "must haves" to cut a release?

**Comment by oliora** [comment]
From my side, I would like to have at least the following issues resolved before release:
* #100 - Socket should have close() method 
* #96 - Confusing usage of ZMQ_RCVMORE flag 
* #77 - Destruction of socket doesn't cancel async operations 
* #46 - non-throwing APIs should be marked noexcept bug 
* #37 - Outdated FindAzioMQ.cmake (at least a simple version of package config)

**Comment by rodgert** [comment]
I would agree with this list, #77 and #100 can be more or less handled in
the same fix. I will also try to go through and kill off the remaining
vestiges of #96 at the same time.

**Comment by oliora** [comment]
Then I can take the rest #46 and #37 :)

Having a boost free version is also nice to have btw but it looks like there is still a long way to achieve this.

**Comment by ahundt** [comment]
Just wanted to bump this, would like to incorporate an azmq build into homebrew.

**Comment by rodgert** [comment]
I have a can't move deliverable for Nov 2, I won't have time until then to
focus on this :(

**Comment by oliora** [comment]
I'm going to spend some time at the end of October for this

**Comment by ahundt** [comment]
Thanks!

**Comment by rodgert** [comment]
I've finally gotten around to doing this. There's now a top level azmq1-0.

@oliora would it be possible to update the CI builds at some point to factor this in?

**Comment by ovanes** [comment]
I see that this issue was closed. Among other issues it references issues: #37, #46 and #46 which are stated to required for the release, but not closed. Are these issues fixed or not? It looks a bit misleading in the corresponding issue to have a line:
> `oliora referenced this issue on 12 Aug 2015`
> `azmq release #101                   Closed`

Is the documentation stating ZeroMQ dependency to be `ZeroMQ 4.0.x` still the only valid option, or are `ZeroMQ 4.1.x - 4.2.x` versions supported as well?

**Comment by rodgert** [comment]
I closed those issues because I'm currently reviewing the overall direction
for a subsequent update to azmq, and I am going to start creating a set of
new issues for a subsequent release. I agree it's a bit misleading, but
they were not going to get fixed before cutting a 1.0, and may not be
addressed as stated in the original issue.

I will make sure they are revisited by end of week, and reopen or create
anew, accordingly.

On a similar note, 4.0.x is the only version I've officially tested
against, but I also plan on supporting 4.1.x for a new release. I don't
anticipate any major problems, as it should mostly 'just work', but there
were contributions I made to libzmq to make some things more streamlined to
implement in azmq, and I'd like to leverage those features, conditionally
on 4.1.x

Tom.
